docs/snapshots.rst

```diff
@@ -91,3 +93,33 @@ they next poll our mutable directory.
 We cannot sign local Snapshots because we don't yet have the
 capability-string for the content. The capability-string is only known
 after we upload the content to Tahoe-LAFS.
+
+
+Signature Details
+-----------------
+
+As above, we upload the content and metadata as two distinct immutable
+capability. There is also the "mangled" name of the file. This mangled
+name will also point at the Snapshot from the Personal DMD.
+
+The signature scheme is to concatenate all three of the above
+(separated by newlines) and then produce a signature over it using the
+author's signing key (which is a `nacl.signing.SigningKey` from the
```

Comment:
My cryptographic question here is whether this is what `nacl.signing.SigningKey` is for.  It *seems* like it is.  https://pynacl.readthedocs.io/en/latest/signing/ says "A valid digital signature gives a recipient reason to believe that the message was created by a known sender such that they cannot deny sending it (authentication and non-repudiation) and that the message was not altered in transit (integrity)." which I think is what you want from a signature system.  It would be good to confirm this understanding somehow.

Reply:
Okay I have another cryptographic question, actually.  Should we tag the parts of the value to be signed?  eg

```
content-cap:<content cap>
metadata-cap:<metadata cap>
path:<path>
```

I have some vague idea about the purpose of tagging ... it prevents accidental collisions between different "kinds" of strings that might accidentally have the same value sometimes?  I'm not sure if that's actually true or not, it's just what's in my head.

I'm not exactly sure why it is important here either, except maybe that it somehow does the thing I described above in some circumstance.